.readthedocs.yaml

```diff
@@ -1,12 +1,12 @@
 # Config for building https://tmt.readthedocs.io/
 version: 2
+python:
+  install:
+    - method: pip
+      path: .
+      extra_requirements:
+        - docs
 build:
-    os: ubuntu-22.04
-    tools:
-        python: "3"
-    # Override RTD's default build commands, install hatch & delegate
-    # the rest to our Makefiles.
-    commands:
-      - cat docs/conf.py
-      - pip install hatch
-      - make BUILDDIR=$READTHEDOCS_OUTPUT docs
```

Comment:
Actually, I'd prefer to be explicit here, I don't like the idea of going behind the back of `hatch run docs:html`. I know it runs a very similar command line, but that may change. If there's a command to build tmt docs website, an entry point so to speak, I'd rather stick to it than rely on the possibility of getting the same outcome when running `sphinx-build` directly, as of now, in this environment. Feels like something that can easily break.

Plus, this setup will be harder to maintain. Adding one extra argument to `shipnx-build` in the `html` hatch script might require a follow-up update of this file, or maybe not - but the thing is, who will remember that changing `sphinx-build` in `pyproject.toml` might affect `.readthedocs.yaml`? :) Calling `hatch run docs:html` (or `make docs`) from every place where docs are expected to be built is simply easier to maintain as it's just a single point to modify when necessary.

Reply:
Hmm, one thing to consider is that RTD does append its own flags when it builds with the built-in sphinx support https://readthedocs.org/projects/tmt/builds/22586697/. Might be good to replicate those as both rtd and aphinx environment evolve. Among the current flags used `-E`, `-d`, and `-D` seem nontrivial.

Other than that, yeah, I agree it should be centralized to a single approach.

Reply:
Did the explicit part already land? It seems the conversation has been resolved, but `build` is still depending on implicit commands used by RTD rather than our `hatch` script.

Reply:
Yes, in this case we are using RTD's default scripts. This is because:
- even in the `hatch` script we are not doing anything fancy anymore
- RTD does add some of its own flags which seem nontrivial.
  I've recently cheeked up on `sphinx-tippy` and `sphinx-hoverxref` which use RTD API, but I'm not sure how rtd itself gets the project's metadata, if it's from `objects.inv` or metadata from here. Basically we might be overlooking some of these nuances that other consuming projects might rely on
- RTD setup might evolve in the future, and having it like this, we don't need to worry about how they're doing

Reply:
@happz maybe this needs a revisit?

Reply:
Ok, let's rely on default script.

Can I at least get a comment to `pyproject.toml`, a note above the `html` script explaining it's **not** the one used to build the online, official docs? Because sooner or later the time will come when I forget about the reason, and the fact `hatch run html` is not building the same docs, and I might try to do something about it. A comment would keep this reasoning in code, not just in this discussion.

Reply:
Kay comment added